dev/tasks/python-wheels/github.osx.amd64.yml

```diff
@@ -115,3 +120,52 @@ jobs:
 
       {{ macros.github_upload_releases("arrow/python/repaired_wheels/*.whl")|indent }}
       {{ macros.github_upload_gemfury("arrow/python/repaired_wheels/*.whl")|indent }}
+
+  test:
+    {% raw %}
+    name: "Test wheel for macOS on ${{ matrix.runs-on }} (USE_CONDA=${{ matrix.use-conda }})"
+    {% endraw %}
+    needs:
+      - build
+    {% raw %}
+    runs-on: ${{ matrix.runs-on }}
+    {% endraw %}
+    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on:
+          - macos-11
+          - macos-12
+        use-conda:
+          - "0"
+          - "1"
```

Comment:
Testing with conda doesn't sound very useful. People using conda are supposed to use the conda packages, not wheels.

Reply:
OK. I'll remove conda patterns from our wheel verification nightly jobs too.